src/client/eppo-client.ts

```diff
@@ -351,6 +358,90 @@ export default class EppoClient implements IEppoClient {
     );
   }
 
+  public getBanditAction(
+    flagKey: string,
+    subjectKey: string,
+    subjectAttributes: Attributes,
+    actions: Record<string, Attributes>,
+    defaultValue: string,
+  ): { variation: string; action: string | null } {
+    let variation = this.getStringAssignment(flagKey, subjectKey, subjectAttributes, defaultValue);
+    let action: string | null = null;
+    const banditKey = variation;
+    try {
```

Comment:
`getStringAssignment()` has its own top-level `try-catch,` so we need another one for our bandit logic that takes place afterwards.